module/data/shared/target-field.mjs

```diff
@@ -76,25 +80,39 @@ export default class TargetField extends SchemaField {
       this.target.template.height = null;
     }
 
-    if ( labels ) {
-      const parts = [];
+    const pr = getPluralRules();
 
-      if ( this.target.template.type ) {
-        if ( this.target.template.count > 1 ) parts.push(`${this.target.template.count} ×`);
-        parts.push(this.target.template.size);
-        if ( this.target.template.units in CONFIG.DND5E.movementUnits ) {
-          parts.push(game.i18n.localize(`DND5E.Dist${this.target.template.units.capitalize()}Abbr`));
-        }
-        parts.push(CONFIG.DND5E.areaTargetTypes[this.target.template.type]?.label);
+    // Generate the template label
+    const templateConfig = CONFIG.DND5E.areaTargetTypes[this.target.template.type];
+    if ( templateConfig ) {
+      const parts = [];
+      if ( this.target.template.count > 1 ) parts.push(`${this.target.template.count} ×`);
+      parts.push(this.target.template.size);
+      if ( this.target.template.units in CONFIG.DND5E.movementUnits ) {
+        // TODO: Switch to properly localized measurements with https://github.com/foundryvtt/dnd5e/issues/3958
+        parts.push(game.i18n.localize(`DND5E.Dist${this.target.template.units.capitalize()}Abbr`));
       }
+      this.target.template.label = game.i18n.format(
+        `${templateConfig.counted}.${pr.select(this.target.template.count || 1)}`, { number: parts.filterJoin(" ") }
+      ).trim().capitalize();
+    } else this.target.template.label = "";
 
-      else if ( this.target.affects.type ) {
-        if ( this.target.affects.scalar ) parts.push(this.target.affects.count);
-        parts.push(CONFIG.DND5E.individualTargetTypes[this.target.affects.type]);
-      }
+    // Generate the affects label
+    const affectsConfig = CONFIG.DND5E.individualTargetTypes[this.target.affects.type];
+    this.target.affects.labels = {
+      sheet: affectsConfig?.counted ? game.i18n.format(
+        `${affectsConfig.counted}.${this.target.affects.count ? pr.select(this.target.affects.count) : "other"}`, {
+          number: this.target.affects.count ? formatNumber(this.target.affects.count)
+            : game.i18n.localize(`DND5E.TARGET.Count.${this.target.template.type ? "Every" : "Any"}`)
+        }
+      ).trim().capitalize() : (affectsConfig?.label ?? ""),
+      statblock: game.i18n.format(
+        `${affectsConfig?.counted ?? "DND5E.TARGET.Type.Target.Counted"}.${pr.select(this.target.affects.count || 1)}`,
```

Comment:
Do we need to handle the case of `self` here, which doesn't have a `counted`? Maybe not for NPC stat blocks?

Reply:
Haven't seen any attacks that target self yet, maybe just a wait-and-see.

Reply:
Yeah, fair enough